Documentation/dev-guide/api_grpc_gateway.md

```diff
@@ -0,0 +1,40 @@
+
+## Why grpc-gateway
+
+etcd v3 uses [gRPC][grpc]. Most recommended is to use etcd [Go client][go-client] and [etcdctl][etcdctl]. However, some languages are not supported by gRPC. This is where [grpc-gateway][grpc-gateway] steps in. It builds a RESTful proxy based on the gRPC services, which can serve HTTP and JSON requests.
+
+
+## How grpc-gateway
+
+The gateway accepts the [JSON mapping][json-mapping] for the messages defined in [Protocol Buffer][api-ref] format. Note that key, value are defined as bytes in Protocol Buffer, thus need to be encoded as base64 string.
+
+```bash
+<<COMMENT
+https://www.base64encode.org/
+foo is 'Zm9v' in Base64
+bar is 'YmFy'
+COMMENT
+
+curl -L http://localhost:2379/v3alpha/kv/put \
+	-X POST -d '{"key": "Zm9v", "value": "YmFy"}'
+
+curl -L http://localhost:2379/v3alpha/kv/range \
+	-X POST -d '{"key": "Zm9v"}'
+
+ETCDCTL_API=3 etcdctl --endpoints=localhost:2379 get foo
```

Comment:
I put it there to show that it was successfully written. But now it looks un-related. Will remove.